fiftyone/core/odm/document.py

```diff
@@ -440,14 +458,15 @@ def from_dict(cls, d, extended=False):
                 # extended form
 
                 # pylint: disable=no-member
+                d = cls._simple_filter(d)
                 return cls._from_son(d)
             except Exception:
                 pass
 
         # Construct any necessary extended JSON components like ObjectIds
         # @todo can we optimize this?
         d = json_util.loads(json_util.dumps(d))
-
+        d = cls._simple_filter(d)
```

Comment:
In general, list fields can contain `None` values. For example:

```py
import fiftyone as fo

sample = fo.Sample(filepath="image.jpg", list_field=[None, None])

dataset = fo.Dataset()
dataset.add_sample(sample)

sample.reload()

assert sample.list_field == [None, None]
```

So we can't change `MongoEngineBaseDocument` as this is the base class for all documents, eg sample and frame documents, not just `DatasetDocument`.

A more robust patch here would be to gracefully catch validation errors related to `sample_fields` or `frame_fields` being None in a `DatasetDocument.from_dict()` super method, and, if found, immediately update these field(s) in the database so this patch doesn't get triggered permanently thereafter.

Ideally we'd still have a hypothesis for why this is happening and confirmation that this won't just lead to a downstream error as I described earlier:

> For example, I bet that updating this PR to gracefully omit dataset._doc.sample_fields[i] if None would not actually do much good. It would allow load_dataset() to work, but then I bet the user would immediately get a "found unknown field 'foo'" error upon calling sample = dataset.first() because whatever the None value was we omitted from the schema was probably supposed to be defining an actual field.

Reply:
I see, now it all makes sense after digging into this code! We can certainly do both, with both patching for graceful handling and trying to figure out why `dataset._doc.sample_fields[i]` contains None value sometimes. 

@ehofesmann : do you know if the customers encounter any other problem after removing `null` values from `sample_fields`? Do they do any additional update to the record?

Reply:
I don't think it's correct to assume that a none value field is a bug. This problem should be handled so None doesn't throw an error/is handled appropriately where the error is being thrown and not messing with the schema

Reply:
@kaixi-wang agreed with the argument from Brian above that filtering out None value is not the solution. In general, do we have a sense when `None` value is acceptable as a value and when it is not? Pymongo throws an exception here because it is not accepting `None` as a value, so in this case the conversion fails.